src/ucp/core/ucp_worker.c

```diff
@@ -1358,21 +1370,13 @@ ucs_status_t ucp_worker_iface_open(ucp_worker_h worker, ucp_rsc_index_t tl_id,
         goto err_close_iface;
     }
 
-    ucp_worker_get_sys_device_distance(context, wiface->rsc_index,
-                                       &wiface->distance);
+    ucp_worker_get_sys_device_distance(wiface);
     if (!context->config.ext.proto_enable) {
-        status = ucp_worker_get_sys_device_distance(context, wiface->rsc_index,
-                                                    &distance);
-        if (status == UCS_OK) {
-            ucp_worker_iface_add_distance(&wiface->attr, &distance);
-        }
+        ucp_worker_iface_add_distance(&wiface->attr, &wiface->distance);
     }
 
-    sys_device = context->tl_rscs[wiface->rsc_index].tl_rsc.sys_device;
-    status     = ucs_topo_get_memory_distance(sys_device, &distance);
-    if (status == UCS_OK) {
-        ucp_worker_iface_add_distance(&wiface->attr, &distance);
-    }
+    ucp_worker_get_sys_device_memory_distance(wiface);
+    ucp_worker_iface_add_distance(&wiface->attr, &wiface->memory_distance);
```

Comment:
but ucp_worker_iface_estimate_perf still calls UCT estimate perf
and ucs_topo_get_memory_distance should be quite fast since the NUMA distances are saved in a hash in ucs/topo

Reply:
This is true. Will revert.